graphql_compiler/query_pagination/__init__.py

```diff
@@ -10,22 +10,22 @@
 from .typedefs import PageAndRemainder
 
 
-def _estimate_number_of_pages(query: ASTWithParameters, result_size: float, page_size: int) -> int:
+def _estimate_number_of_pages(
+    query: QueryStringWithParameters, result_size: float, page_size: int
+) -> int:
     """Estimate how many pages of results we should generate to meet the desired result size.
 
     Args:
-        query: ASTWithParameters
-        result_size: The estimated result size of a query
-        page_size: The desired page size of a query
+        query: query string and parameters being analyzed for pagination
+        result_size: estimated result size of running the supplied query as-is
+        page_size: desired page size of the final paginated query
 
     Returns:
-        int, estimated number of pages if the query were executed.
-
-    Raises:
-        ValueError if page_size is below 1.
+        int, estimated number of pages of the desired size into which the supplied query
+        should be split up
     """
     if page_size < 1:
-        raise ValueError(
+        raise AssertionError(
```

Comment:
This is a trivial invariant for the caller to enforce. `ValueError` made it the callee's problem, `AssertionError` makes it the caller's problem, which I believe is better since the caller is better positioned to produce a meaningful error message if this were to ever legitimately occur.